src/workers/personalizedDigestEmail.ts

```diff
@@ -0,0 +1,165 @@
+import { In } from 'typeorm';
+import {
+  baseNotificationEmailData,
+  feedToFilters,
+  getDiscussionLink,
+  pickImageUrl,
+  sendEmail,
+} from '../common';
+import { ArticlePost, User } from '../entity';
+import { UserPersonalizedDigest } from '../entity/UserPersonalizedDigest';
+import { messageToJson, Worker } from './worker';
+import { DayOfWeek } from '../types';
+import { MailDataRequired } from '@sendgrid/mail';
+import { format, nextDay, previousDay } from 'date-fns';
+import { zonedTimeToUtc } from 'date-fns-tz';
+import { personalizedDigestFeedClient } from '../integrations/feed/generators';
+
+interface Data {
+  personalizedDigest: UserPersonalizedDigest;
+}
+
+const personalizedDigestPostsCount = 5;
+
+const emailTemplateId = 'd-328d1104d2e04fa1ab91e410e02751cb';
+
+const personalizedDigestDateFormat = 'yyyy-MM-dd HH:mm:ss';
+
+const getEmailSendDate = ({
+  personalizedDigest,
+}: {
+  personalizedDigest: UserPersonalizedDigest;
+}): Date => {
+  const nextPreferredDay = nextDay(new Date(), personalizedDigest.preferredDay);
+  nextPreferredDay.setHours(personalizedDigest.preferredHour, 0, 0, 0);
+  const sendDateInPreferredTimezone = zonedTimeToUtc(
+    nextPreferredDay,
+    personalizedDigest.preferredTimezone,
+  );
+
+  return sendDateInPreferredTimezone;
+};
+
+const getPreviousSendDate = ({
+  personalizedDigest,
+}: {
+  personalizedDigest: UserPersonalizedDigest;
+}): Date => {
+  const nextPreferredDay = previousDay(
+    new Date(),
+    personalizedDigest.preferredDay,
+  );
+  nextPreferredDay.setHours(personalizedDigest.preferredHour, 0, 0, 0);
+  const sendDateInPreferredTimezone = zonedTimeToUtc(
+    nextPreferredDay,
+    personalizedDigest.preferredTimezone,
+  );
+
+  return sendDateInPreferredTimezone;
+};
+
+const getPostsTemplateData = async ({ posts }: { posts: ArticlePost[] }) => {
+  const templateData = await Promise.all(
+    posts.map(async (post) => {
+      const source = await post.source;
+
+      const postUrl = new URL(getDiscussionLink(post.id));
+      postUrl.searchParams.append('utm_source', 'notification');
+      postUrl.searchParams.append('utm_medium', 'email');
+      postUrl.searchParams.append('utm_campaign', 'digest');
+
+      return {
+        post_title: post.title,
+        post_image: post.image || pickImageUrl(post),
+        post_link: postUrl.toString(),
+        source_name: source.name,
+        source_image: source.image,
+      };
+    }),
+  );
+
+  return templateData;
+};
+
+const worker: Worker = {
+  subscription: 'api.personalized-digest-email',
+  handler: async (message, con, logger) => {
+    if (process.env.NODE_ENV === 'development') {
+      return;
```

Comment:
Not sure how to mock this locally, `http://localhost:6000/feed.json` used for feeds is not available. Any recommendations?

Reply:
You disabled it for the local env so it will not run?